### PR TITLE
Add `pulp_href__in` and `pulp_id__in` query filters

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #2975

@daviddavis Would this work for you? These filters will be added to each endpoint.

### CODE REVIEW

**Comment by daviddavis** [comment]
The values for these fields, are they json lists or comma-separated? It looks like the former if my reading of the code is correct.

**Comment by daviddavis** [comment]
Oh actually I just read the docs for `BaseInFilter` and it's comma separated which is probably easier for us.

This looks great, thanks.

**Review by ggainey — APPROVED** [review]

**Review by decko — APPROVED** [review]

File: pulpcore/filters.py [hunk 1]
```diff
@@ -164,6 +176,11 @@ class BaseFilterSet(filterset.FilterSet):
         "ne": _("not equal to"),
     }
 
+    def filter_pulp_href(self, queryset, name, value):
+        # Convert each href to a pk
+        pks = [extract_pk(href) for href in value]
```
Comment (mdellweg) [inline pulpcore/filters.py:181]: This looks like allowing anything that looks like an HREF containing a uuid to be used. Does it work with repository versions?

Comment (gerrod3) [reply pulpcore/filters.py:181]: You are correct, this did not work with repository versions. Fix here: https://github.com/pulp/pulpcore/pull/3733

**Pull request merged**